### PR TITLE
Option Static routes support - RFC 3442

### PR DESCRIPTION
**Pull request by davidblaisonneau-orange**
This PR add an option to add static routes with DHCP.
See:
 * https://tools.ietf.org/html/rfc3442
 * https://ercpe.de/blog/pushing-static-routes-with-isc-dhcp-server

### CODE REVIEW

File: templates/dhcpd.conf.erb [hunk 1]
```diff
@@ -34,6 +34,11 @@ option fqdn.no-client-update    on;  # set the "O" and "S" flag bits
 option fqdn.rcode2            255;
 option pxegrub code 150 = text ;
 
+<% if has_variable?( 'option_static_route' ) -%>
```
Comment (ekohl) [inline templates/dhcpd.conf.erb]: I think you also need a check if it's non-empty.

File: templates/dhcpd.pool.erb [hunk 2]
```diff
@@ -16,6 +16,12 @@ subnet <%= @network %> netmask <%= @mask %> {
 <% if @gateway && !@gateway.strip.empty? -%>
   option routers <%= @gateway %>;
 <% end -%>
+<% if @static_routes
+  @static_routes.each do |static_route| -%>
+  option rfc3442-classless-static-routes <%= static_route['mask'] %>, <%= static_route['network'].split('.').join(', ') %>, <%= static_route['gateway'].split('.').join(', ') %>;
+  option ms-classless-static-routes  <%= static_route['mask'] %>, <%= static_route['network'].split('.').join(', ') %>, <%= static_route['gateway'].split('.').join(', ') %>;
+  <% end
+end -%>
```
Comment (ekohl) [inline templates/dhcpd.pool.erb]: This is a weird end statement.

**Comment by ekohl** [comment]
Would you mind adding a test case as well? And the tests need to pass. Otherwise I think it's a good addition.

File: templates/dhcpd.pool.erb (hunk 2 above)
Comment (davidblaisonneau-orange) [inline templates/dhcpd.pool.erb]: I can rewrite it if you prefer, but it will add an empty line in the file

Comment (ekohl) [inline templates/dhcpd.pool.erb]: Really? I'd expect the following wouldn't:
```erb
<% if @static_routes -%>
<% @static_routes.each do |static_route| -%>
  option rfc3442-classless-static-routes <%= static_route['mask'] %>, <%= static_route['network'].split('.').join(', ') %>, <%= static_route['gateway'].split('.').join(', ') %>;
  option ms-classless-static-routes  <%= static_route['mask'] %>, <%= static_route['network'].split('.').join(', ') %>, <%= static_route['gateway'].split('.').join(', ') %>;
<% end -%>
<% end -%>
```

But I didn't test it.

**Comment by davidblaisonneau-orange** [comment]
Travis seems to be happy :) I hope the tests are ok for you.

**Comment by ekohl** [comment]
:+1: 
@domcleal / @mmoll?

**Comment by mmoll** [comment]
I don't use the -dhcp module, so I can't say too much content-wise, but the tests look OK, so :+1: 

@davidblaisonneau-orange could you squash your commits together into one?

**Comment by mmoll** [comment]
@davidblaisonneau-orange there are still 8 commits. ```git rebase -i HEAD~8``` and the usual squash/fixup dance should do the trick, followed by a force push into the master branch of your fork.

**Comment by ekohl** [comment]
BTW, I don't mind squashing it and pushing the squashed commit if you can't figure it out.

**Comment by davidblaisonneau-orange** [comment]
This should be better. Sorry for the delay, PR is something quite new for me

**Comment by ekohl** [comment]
@davidblaisonneau-orange no problem, we've all been there. The only thing that would finish it if you could fix the commit message to something more meaningful.

**Comment by davidblaisonneau-orange** [comment]
Now it should be fine :)

**Comment by ekohl** [comment]
:+1:

File: templates/dhcpd.pool.erb [hunk 3]
```diff
@@ -16,6 +16,12 @@ subnet <%= @network %> netmask <%= @mask %> {
 <% if @gateway && !@gateway.strip.empty? -%>
   option routers <%= @gateway %>;
 <% end -%>
+<% if @static_routes
+  @static_routes.each do |static_route| -%>
```
Comment (mmoll) [inline templates/dhcpd.pool.erb]: could you please write
```erb
<% if @static_routes -%>
<%   @static_routes.each do |static_route| -%>
...
<%   end -%>
<% end -%>
```
or similar here, I was just confused by seeing to ```<% end -%>```s and thought of a duplicate line.

**Comment by mmoll** [comment]
:+1: after fixing the minor whitespace nit.

**Comment by davidblaisonneau-orange** [comment]
Done

**Comment by mmoll** [comment]
:+1:

**Comment by ekohl** [comment]
Thanks!

**Comment by davidblaisonneau-orange** [comment]
Thanks to you !

**Pull request closed without merging**